app/resources/admin_api_key_form.html.template

```diff
@@ -14,6 +14,16 @@
   function $(id) {
     return document.getElementById(id);
   }
+  function updateWriteDomain() {
```

Comment:
Can you clear domain_write_field when the checkbox is unchecked? Otherwise, I believe, if the user fills in the domain name, uncheck the checkbox and submit, then the API key will end up having write permission.

Also, can you rename this to update_write_domain to be consistent with other method names?